src/addons/addon.h

```diff
@@ -27,7 +31,11 @@ class Addon : public QObject {
 
   static bool evaluateConditions(const QJsonObject& conditions);
 
+  enum State { Installed, Enabled, Disabled };
```

Comment:
Maybe you can do the following:
1. Add an Unknown state and set it as default in m_state
2. In the CTOR of the addon class, you read the state from the settings. If you don't find it, call updateState(installed)
3. Remove the change in storeAndLoad

It should work. What do you think? In this way, it does not matter why and when we create the addon obj. At the first creation we set it as installed.

If we do this, we can add an assertion in updateState to see if the new state != Unknown.